src/Data/Slug.purs

```diff
@@ -0,0 +1,120 @@
+module Data.Slug
+  ( Slug
+  , generate
+  , parse
+  , toString
+  , truncate
+  ) where
+
+import Prelude
+
+import Data.Argonaut.Decode (class DecodeJson, JsonDecodeError(..), decodeJson)
+import Data.Argonaut.Encode (class EncodeJson, encodeJson)
+import Data.Array as Array
+import Data.CodePoint.Unicode (isLatin1)
+import Data.Either (note)
+import Data.Maybe (Maybe(..))
+import Data.String as String
+import Data.String.CodePoints (codePointFromChar, fromCodePointArray, toCodePointArray)
+import Data.String.Pattern (Pattern(..), Replacement(..))
+
+-- | A `Slug` represents a string value which is guaranteed to have the
+-- | following qualities:
+-- |
+-- | - it is not empty
+-- | - it consists of groups of characters separated by `-` dashes,
+-- |   where the slug cannot begin or end with a dash, and there can
+-- |   never be two dashes in a row.
+-- |
+-- | Example: `Slug "this-is-an-article-slug"`
+newtype Slug = Slug String
+
+derive newtype instance Eq Slug
+derive newtype instance Ord Slug
+derive newtype instance Semigroup Slug
+
+instance Show Slug where
+  show (Slug str) = "(Slug " <> str <> ")"
```

Comment:
I think we should ditch these instances since they aren’t used here (I generally don’t use Show, and we use codecs instead of JSON instances)

Reply:
I removed the `*Json`, `Show`, and `Semigroup` instances, `Eq` and `Ord` are used in `instance ordRoute :: Ord Route`.